### DISCUSSION TITLE
Enum has unconstructable fields

### DISCUSSION THREAD

**Discussion by Novakasa** [jakobhellermann/bevy-inspector-egui#146] [Q&A] [answered]
Hi all, I am rather inexperienced in rust still, so I am not sure whether this is a limitation of the crate or whether I am doing something wrong. I have this structure for the `StepKey` type:

```rust

#[derive(
    Debug, Clone, Copy, PartialEq, Eq, Hash, Display, Reflect, Serialize, Deserialize, Default,
)]
#[reflect(Default)]
enum StepTFKey {
    Decel,
    #[default]
    Neutral,
    Accel,
}

#[derive(
    Debug, Clone, Copy, PartialEq, Eq, Hash, Display, Reflect, Serialize, Deserialize, Default,
)]
#[reflect(Default)]
enum StepNFKey {
    Low,
    #[default]
    Medium,
    High,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Reflect, Serialize, Deserialize, Default)]
#[reflect(Default)]
enum StepKey {
    Run(StepTFKey, StepNFKey),
    #[default]
    Walk,
}

```

Now when I try to display a `StepKey` in a ResourceInspector, the `Run` variant is unselectable with the following tooltip:
![image](https://github.com/jakobhellermann/bevy-inspector-egui/assets/59559612/a9ce69c8-22b9-4765-aeb0-b85babda5265)

As you can see in the snippet, I already added `#[reflect(Default)]` to all types. Is it impossible to display this kind of structure in bevy_inspector_egui?

**Comment by rlidwka** [comment, marked as answer]
I'm not completely sure, but maybe `StepNFKey` is not registered? Make sure `app.register_type()` is called on all three.

Also, can you provide a full code snippet that we can run?

**Reply by Novakasa** [reply]
> I'm not completely sure, but maybe `StepNFKey` is not registered? Make sure `app.register_type()` is called on all three.

Oh man, that indeed was my issue. I even stumbled on this before, so that's embarrasing. I guess the mention of `reflect(Default)` really threw me off. Thanks for the help!
